ractor.rb

```diff
@@ -153,26 +512,60 @@ class RemoteError
   end
 
   # Closes the incoming port and returns its previous state.
+  # All further attempts to Ractor.receive in the ractor, and #send to the ractor
```

Comment:
I think it is valuable that incoming/outgoing ports are closed automatically when the ractor terminates.

more details:
on closed incoming port, `Ractor#send` will raise immediately.
But `Ractor.receive` doesn't raise an error until there are sent messages in incoming queue.
if the queue is empty, `Ractor.receive` raises.

outgoing port will be closed after the last yielded message (ractor's block value or exception which terminates the ractor) is taken.

Reply:
I too believe it is important, and explain this in docs of `#send` and and `#take`, I believe it belongs there, not to the method for explicitly closing ports, don't you think?..

> But `Ractor.receive` doesn't raise an error until there are sent messages in incoming queue.

Hm, in my experiment it raises immediately on closed incoming:

```ruby
Ractor.new { 
  close_incoming
  receive 
}
# #<Thread:0x00005652ebd41ae0 run> terminated with exception (report_on_exception is true):
# <internal:ractor>:98:in `receive': The incoming port is already closed (Ractor::ClosedError)
```

Reply:
> I too believe it is important, and explain this in docs of #send and and #take, I believe it belongs there, not to the method for explicitly closing ports, don't you think?..

Ah... I think automatic closing should be also explained here (reference to the details is enough).

> Hm, in my experiment it raises immediately on closed incoming:

```ruby
r = Ractor.new Ractor.current do |main|
  main << true
  p receive #=> 1 (closed, but remained)
  p receive #=> 2 (closed, but remained)
  p receive #=> The incoming port is already closed (Ractor::ClosedError)
end

r << 1
r << 2
Ractor.receive
r.close_incoming
r.take
```